resource/model_test.go

```diff
@@ -143,56 +143,56 @@ func TestModelFromString(t *testing.T) {
 			"acme/corp:test:modelA",
 			resource.Model{},
 			"not a valid model name",
-			"invalid character",
```

Comment:
For context, these expected errors were the ones returned by `json.Unmarshal` when it was unable to turn `arm/corp:test:modelA` into a model struct. Now, we return a more informative error if model doesn't match valid regexes and doesn't contain an opening brace.